doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1814 @@
+
+=item Combining the Configure options no-ec and no-dh no longer disables TLSv1.3
+
+Typically if OpenSSL has no EC or DH algorithms then it cannot support
+connections with TLSv1.3. However OpenSSL now supports "pluggable" groups
+through providers. Therefore third party providers may supply group
+implementations even where there are no built-in ones. Attempting to create
+TLS connections in such a build without also disabling TLSv1.3 at run time or
+using third party provider groups may result in handshake failures. TLSv1.3
+can be disabled at compile time using the "no-tls1_3" Configure option.
+
+=item SSL_CTX_set_ciphersuites() and SSL_set_ciphersuites() changes.
+
+The methods now ignore unknown ciphers.
+
+=item Security callback change.
+
+The security callback, which can be customised by application code, supports
+the security operation SSL_SECOP_TMP_DH. This is defined to take an EVP_PKEY
+in the "other" parameter. In most places this is what is passed. All these
+places occur server side. However there was one client side call of this
+security operation and it passed a DH object instead. This is incorrect
+according to the definition of SSL_SECOP_TMP_DH, and is inconsistent with all
+of the other locations. Therefore this client side call has been changed to
+pass an EVP_PKEY instead.
+
+=item New SSL option SSL_OP_IGNORE_UNEXPECTED_EOF
+
+The SSL option SSL_OP_IGNORE_UNEXPECTED_EOF is introduced. If that option
+is set, an unexpected EOF is ignored, it pretends a close notify was received
+instead and so the returned error becomes SSL_ERROR_ZERO_RETURN.
+
+=item The security strength of SHA1 and MD5 based signatures in TLS has been reduced.
+
+This results in SSL 3, TLS 1.0, TLS 1.1 and DTLS 1.0 no longer
+working at the default security level of 1 and instead requires security
+level 0. The security level can be changed either using the cipher string
+with `@SECLEVEL`, or calling L<SSL_CTX_set_security_level(3)>. This also means
+that where the signature algorithms extension is missing from a ClientHello
+then the handshake will fail in TLS 1.2 at security level 1. This is because,
+although this extension is optional, failing to provide one means that
+OpenSSL will fallback to a default set of signature algorithms. This default
+set requires the availability of SHA1.
+
+=item X509 certificates signed using SHA1 are no longer allowed at security level 1 and above.
+
+In TLS/SSL the default security level is 1. It can be set either using the cipher
+string with `@SECLEVEL`, or calling L<SSL_CTX_set_security_level(3)>. If the
+leaf certificate is signed with SHA-1, a call to L<SSL_CTX_use_certificate(3)>
+will fail if the security level is not lowered first.
+Outside TLS/SSL, the default security level is -1 (effectively 0). It can
+be set using L<X509_VERIFY_PARAM_set_auth_level(3)> or using the `-auth_level`
+options of the commands.
+
+=back
+
+=head1 RETURN VALUES
+
+See the individual manual pages for details.
```

Comment:
I'm not sure we need this section at all.

Reply:
DOc-nits didnt complain so removed it from fips_module.pod also